storagemarket/provider_test.go

```diff
@@ -49,15 +51,14 @@ func TestSimpleDealHappy(t *testing.T) {
 
 	// setup the provider test harness
 	harness := NewHarness(t, ctx)
-	defer harness.Stop()
```

Comment:
I'm curious why did you change defer harness.Stop() everywhere in this file? My understanding is that if there's a test failure, the test execution will immediately stop. But the defer will cause harness.Stop() to be called even if test execution aborts.

Reply:
Hmm, I don't remember why I did this. I agree, it's wrong. I've put it back now.